docs/source/user-guide/supported_models.rst

```diff
@@ -0,0 +1,65 @@
+.. _supported_models:
+
+====================
+Supported Models
+====================
+
+Similar to scikit-learn API, a model will be compatible with deepchecks' checks if it implements some minimal functionality.
+Without a compatible model, deepchecks can't run all checks that require predictions, for example all :docs:`performance checks </api/checks/performance>`.
+
+Below are the minimal requirements for each task type:
+
+Required API
+=============
+
+Regression
+----------
+
+All that is necessary for a regression model is the ``predict`` function.
+The predict function should expect an `array-like` or `sparse matrix` of shape ``(n_samples, n_features)``
```

Comment:
consider linking to array-like:
https://scikit-learn.org/stable/glossary.html#term-array-like